app/client/lib/UrlState.ts

```diff
@@ -125,17 +125,24 @@ export class UrlState<IUrlState extends object> extends Disposable {
    * Applies to an <a> element to create a smart link, e.g. dom('a', setLinkUrl({ws: wsId})). It
    * both sets the href (e.g. to allow the link to be opened to a new tab), AND intercepts plain
    * clicks on it to "follow" the link without reloading the page.
+   *
+   * If a "beforeChange" option is passed in, it will be run just before changing the URL.
```

Comment:
Can you update this comment with intended use? Like:
`... Primary used for clearing any unsaved state before navigating away`

Reply:
Updated comment